File: gaia-sdk-python/gaia_sdk/graphql/request/type/IdentityMetrics.py

from typing import Callable

from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.type.BehaviourState import BehaviourState
from gaia_sdk.graphql.request.type.IntentDetectionRate import IntentDetectionRate
from gaia_sdk.graphql.request.type.MetricsEntityCount import MetricsEntityCount
from gaia_sdk.graphql.request.type.TopExecutedBehaviour import TopExecutedBehaviour


class IdentityMetrics(list):
    """
    Represents identity metrics information
    """

    def identity_id(self):
        self.append(lambda x: "identityId")

    def entity_count(self, config: Callable[['MetricsEntityCount'], None]):
        def callback(registry: VariableRegistry):
            entity = MetricsEntityCount()
            config(entity)
            return "entity_count {" + entity.render(registry) + "}"
        self.append(callback)

    def top_executed_behaviours(self, config: Callable[['TopExecutedBehaviour'], None]):
        def callback(registry: VariableRegistry):
            entity = TopExecutedBehaviour()
            config(entity)
            return "top_executed_behaviours {" + entity.render(registry) + "}"
        self.append(callback)

    def behaviour_states(self, config: Callable[['BehaviourState'], None]):
        def callback(registry: VariableRegistry):
            entity = BehaviourState()
            config(entity)
            return "behaviour_states {" + entity.render(registry) + "}"
        self.append(callback)

    def intent_detection_rate(self, config: Callable[['IntentDetectionRate'], None]):
        def callback(registry: VariableRegistry):
            entity = IntentDetectionRate()
            config(entity)
            return "intent_detection_rate {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/IdentityMetrics.ts

import {IntentDetectionRate} from "./IntentDetectionRate";
import {TopExecutedBehaviour} from "./TopExecutedBehaviour";
import {BehaviourState} from "./BehaviourState";
import {MetricsEntityCount} from "./MetricsEntityCount";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Represents identity metrics information
 */
export class IdentityMetrics extends Array<(_:VariableRegistry) => string> {
public _typeName = "IdentityMetrics";
    public identityId = () => { 
        this.push(_ => "identityId")
    };

    public entityCount = (config: (_:MetricsEntityCount) => void) => this.push((registry) => {
        const entity = new MetricsEntityCount();
        config(entity);
        return "entityCount { " + entity.render(registry) + " }";
    });

    public topExecutedBehaviours = (config: (_:TopExecutedBehaviour) => void) => this.push((registry) => {
        const entity = new TopExecutedBehaviour();
        config(entity);
        return "topExecutedBehaviours { " + entity.render(registry) + " }";
    });

    public behaviourStates = (config: (_:BehaviourState) => void) => this.push((registry) => {
        const entity = new BehaviourState();
        config(entity);
        return "behaviourStates { " + entity.render(registry) + " }";
    });

    public intentDetectionRate = (config: (_:IntentDetectionRate) => void) => this.push((registry) => {
        const entity = new IntentDetectionRate();
        config(entity);
        return "intentDetectionRate { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

